code/game/objects/machinery/bots/bots.dm

```diff
@@ -1,18 +1,143 @@
 /obj/machinery/bot
+	name = "generic utility robot"
+	desc = "a generic utility robot, ahelp if you see this in game."
 	icon = 'icons/obj/aibots.dmi'
-	layer = MOB_LAYER
-	use_power = NO_POWER_USE
-	resistance_flags = XENO_DAMAGEABLE
-
-/obj/machinery/bot/cleanbot
-	name = "Cleanbot"
-	desc = "A little cleaning robot, he looks so excited!"
-	icon = 'icons/obj/aibots.dmi'
-	icon_state = "cleanbot0"
 	density = FALSE
 	anchored = FALSE
-	max_integrity = 25
+	voice_filter = "alimiter=0.9,acompressor=threshold=0.2:ratio=20:attack=10:release=50:makeup=2,highpass=f=1000"
+	///Keeps track of how many items or whatever have been cleaned
+	var/counter = 0
+	///So It doesnt infinitely look for an exit and crash the server
+	var/stuck_counter = 0
+	///strings used on doing an action like cleaning
+	var/list/sentences = list()
+	///strings used on activating the bot
+	var/list/awakeningsentences = list()
+	///strings used on shutting down the robot
+	var/list/shutdownsentences = list()
+	///is the robot active
+	var/is_active = FALSE
+	///can the robot be turned on or off
+	var/alter_operating_mode = FALSE
+	///animation to play when the robot is started by hand
+	var/activation_animation = null
+	///animation to play when the robot is started by hand
+	var/deactivation_animation = null
+	///icon to set while active
+	var/active_icon_state = null
+
+/obj/machinery/bot/Initialize(mapload)
+	. = ..()
+	if(SStts.tts_enabled)
+		var/static/todays_voice
+		if(!todays_voice)
+			todays_voice = pick(SStts.available_speakers)
+		voice = todays_voice
+	RegisterSignal(src, COMSIG_AREA_EXITED, PROC_REF(turn_around))
+	update_icon_state()
```

Comment:
Don't call update_icon_state directly, use update_icon instead and let it handle state.